plugins/woocommerce/src/Internal/Fulfillments/OrderFulfillmentsRestController.php

```diff
@@ -0,0 +1,1084 @@
+
+		if ( is_null( $error_information ) ) {
+			return false;
+		}
+
+		return new \WP_Error(
+			$error_information['code'],
+			$error_information['message'],
+			array( 'status' => rest_authorization_required_code() )
+		);
+	}
+
+	/**
+	 * Get the fulfillments for the order.
+	 *
+	 * @param WP_REST_Request $request The request object.
+	 *
+	 * @return WP_REST_Response The fulfillments for the order, or an error if the request fails.
+	 */
+	public function get_fulfillments( WP_REST_Request $request ): WP_REST_Response {
+		$order_id     = (int) $request->get_param( 'order_id' );
+		$fulfillments = array();
+
+		// Fetch fulfillments for the order.
+		try {
+			$datastore    = wc_get_container()->get( FulfillmentsDataStore::class );
+			$fulfillments = $datastore->read_fulfillments( WC_Order::class, "$order_id" );
+		} catch ( \Exception $e ) {
+			return $this->prepare_error_response(
+				$e->getCode(),
+				$e->getMessage(),
+				WP_Http::BAD_REQUEST
+			);
+		}
+
+		// Return the fulfillments.
+		return new WP_REST_Response(
+			array(
+				'fulfillments' => array_map(
+					function ( $fulfillment ) {
+						return $fulfillment->get_raw_data(); },
+					$fulfillments
+				),
+			),
+			WP_Http::OK
+		);
+	}
+
+	/**
+	 * Create a new fulfillment with the given data for the order.
+	 *
+	 * @param WP_REST_Request $request The request object.
+	 *
+	 * @return WP_REST_Response The created fulfillment, or an error if the request fails.
+	 */
+	public function create_fulfillment( WP_REST_Request $request ) {
+		$order_id = (int) $request->get_param( 'order_id' );
+
+		// Create a new fulfillment.
+		try {
+			$fulfillment = new Fulfillment();
+			$fulfillment->set_props( $request->get_json_params() );
+			$fulfillment->set_meta_data( $request->get_json_params()['meta_data'] );
+			$fulfillment->set_entity_type( WC_Order::class );
+			$fulfillment->set_entity_id( "$order_id" );
+
+			$fulfillment->save();
+		} catch ( \Exception $e ) {
+			return $this->prepare_error_response(
+				$e->getCode(),
+				$e->getMessage(),
+				WP_Http::BAD_REQUEST
+			);
+		}
+
+		return new WP_REST_Response( array( 'fulfillment' => $fulfillment->get_raw_data() ), WP_Http::CREATED );
+	}
+
+	/**
+	 * Get a specific fulfillment for the order.
+	 *
+	 * @param WP_REST_Request $request The request object.
+	 *
+	 * @return WP_REST_Response The fulfillment for the order, or an error if the request fails.
+	 */
+	public function get_fulfillment( WP_REST_Request $request ): WP_REST_Response {
+		$order_id       = (int) $request->get_param( 'order_id' );
+		$fulfillment_id = (int) $request->get_param( 'fulfillment_id' );
+
+		// Fetch the fulfillment for the order.
+		try {
+			$fulfillment = new Fulfillment( $fulfillment_id );
+			if ( $fulfillment->get_id() !== $fulfillment_id || $fulfillment->get_entity_type() !== WC_Order::class || $fulfillment->get_entity_id() !== "$order_id" ) {
```

Comment:
I see this check repeated multiple times in the controller. It may be worth extracting it since it has various conditions, and it's not super easy to read.

Reply:
Addressed in [a572aef](https://github.com/woocommerce/woocommerce/pull/57496/commits/a572aef1ba164e1091943d467f7b0ea8e567d167).